### PR TITLE
makes the ai detector tool display the vague area the AI is looking at

### LINKED ISSUE #87414: the ai detector thing doesnt display the area ai is looking at
### Client Version:

515.1644

### Issue Summary:

there was supposed to be like a green giant hud thing only you could see that represented where the ai sees

### Round ID:

local?

### Test Merge Information:

idk

### Reproduction Steps:

1. use ai multitool hud

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

https://github.com/user-attachments/assets/3c2b2fd4-6c82-4986-bb90-d9954d48e671

https://github.com/user-attachments/assets/3583f6e6-cc53-4162-a611-5870b938d11e

the ai detector multitool has been slightly redone and the HUD silently removed by #68301 is back
no action button, use inhand to toggle hud

RMB to show static

## Why It's Good For The Game

fixes #87414
also this feature was silently removed

## Changelog
:cl:
fix: brings back the AI detector HUD that was silently removed by a PR 3 years ago, use inhand to toggle hud or use RMB to see static temporarily
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
Good change, I used to use this all the time and wonder why it wasn't fucking doing anything.

File: code/game/objects/items/devices/multitool.dm [hunk 1]
```diff
@@ -94,19 +105,51 @@
...
 /obj/item/multitool/ai_detect/Destroy()
-	STOP_PROCESSING(SSfastprocess, src)
+	if(hud_on && ismob(loc))
+		remove_hud(loc)
 	return ..()
 
-/obj/item/multitool/ai_detect/ui_action_click()
-	return
+/obj/item/multitool/ai_detect/attack_self(mob/user, modifiers)
+	apc_scanner = FALSE //we want to toggle hud, not check for APC
```
Comment (Ghommie) [inline code/game/objects/items/devices/multitool.dm:136]: the apc scanning function should be assigned to one of the two click procs but not both. I'd suggest setting the variable to FALSE on compile and have the action done in attack_self_secondary() instead.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by mc-oofert** [comment]
i still have no clue how to bring back the "see AI static but like almost transparent" part that was removed idk how it works
as far as i can AIs just get images from camera chunks now or something

**Comment by mc-oofert** [comment]
actually im about to cook hold on

**Comment by mc-oofert** [comment]
https://github.com/user-attachments/assets/3583f6e6-cc53-4162-a611-5870b938d11e

new thing

**Pull request merged**